src/metabase/embed/settings.clj

```diff
@@ -5,24 +5,31 @@
    [crypto.random :as crypto-random]
    [metabase.analytics.snowplow :as snowplow]
    [metabase.models.setting :as setting :refer [defsetting]]
+   [metabase.public-settings.premium-features :as premium-features]
    [metabase.util.embed :as embed]
    [metabase.util.i18n :as i18n :refer [deferred-tru]]
+   [metabase.util.malli :as mu]
    [toucan2.core :as t2]))
 
-(defsetting embedding-app-origin
-  (deferred-tru "Allow this origin to embed the full Metabase application")
-  :feature    :embedding
-  :export?    false
-  :visibility :public
-  :audit      :getter)
+;; embedding-app-origin is required by make-embedding-toggle-setter (and vice versa)
+(declare embedding-app-origin)
 
-(defsetting embedding-app-origins-sdk
-  (deferred-tru "Allow this origin to embed Metabase SDK")
-  :feature    :embedding-sdk
-  :export?    false
-  :visibility :public
-  :encryption :never
-  :audit      :getter)
+(mu/defn- make-embedding-toggle-setter
+  "Creates a boolean setter for various boolean embedding-enabled flavors, all tracked by snowplow."
+  [setting-key :- :keyword event-name :- :string]
+  (fn [new-value]
+    (when (not= new-value (setting/get-value-of-type :boolean setting-key))
+      (setting/set-value-of-type! :boolean setting-key new-value)
+      (when (and new-value (str/blank? (embed/embedding-secret-key)))
+        (embed/embedding-secret-key! (crypto-random/hex 32)))
+      (snowplow/track-event! ::snowplow/embed_share
+                             {:event  (keyword (str event-name "-" (if new-value "enabled" "disabled")))
+                              :embedding-app-origin-set   (boolean (embedding-app-origin))
```

Comment:
nit - could use `get-value-of-type` here too to avoid the having to `declare` the setting up top